### PR TITLE
[SPARK-35753][R][INFRA] Increase stacksize in Maven for AppVeyor build

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to increase the stacksize in AppVeyor build. Currently, AppVeyor build is flaky:

```
java.lang.StackOverflowError
scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
scala.reflect.internal.Trees.$anonfun$itransform$1(Trees.scala:1421)
scala.tools.nsc.transform.TypingTransformers$TypingTransformer.atOwner(TypingTransformers.scala:43)
rmer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1409)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
```

https://ci.appveyor.com/project/ApacheSoftwareFoundation/spark

### Why are the changes needed?

To make the tests not flaky.

### Does this PR introduce _any_ user-facing change?
No, dev-only.

### How was this patch tested?
It will be tested via AppVeyor CI in this PR build.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2653,7 +2653,7 @@
               <include>**/*Suite.java</include>
             </includes>
             <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
-            <argLine>-ea -Xmx4g -Xss4m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
+            <argLine>-ea -Xmx4g -Xss64m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```
Comment (HyukjinKwon) [inline pom.xml:2656]:
```suggestion
            <argLine>-ea -Xmx4g -Xss4m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```

File: pom.xml [hunk 2]
```diff
@@ -2704,7 +2704,7 @@
             <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
             <junitxml>.</junitxml>
             <filereports>SparkTestSuite.txt</filereports>
-            <argLine>-ea -Xmx4g -Xss4m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
+            <argLine>-ea -Xmx4g -Xss64m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```
Comment (HyukjinKwon) [inline pom.xml:2707]:
```suggestion
            <argLine>-ea -Xmx4g -Xss4m -XX:ReservedCodeCacheSize=${CodeCacheSize} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```

File: appveyor.yml [hunk 3]
```diff
@@ -57,6 +57,7 @@ environment:
   # "(converted from warning) unable to identify current timezone 'C':" for an unknown reason.
   # This environment variable works around to test SparkR against a higher version.
   R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+  MAVEN_OPTS: -Xss256m
```
Comment (srowen) [inline appveyor.yml:60]: 256MB as a thread stack size sounds very large - the default is 1MB. Does it need to be that high?

Comment (HyukjinKwon) [reply appveyor.yml:60]: Maybe not .. let me debug a bit more.

**Comment by HyukjinKwon** [comment]
I'm gonna drop this PR for now. Changing stacksize apparently doesn't affect the build being flaky.

**Pull request closed without merging**